### PR TITLE
Sliding Window Algorithem

### PR DESCRIPTION
**Pull request by morMBen**
I have now used Standard.js
Hope everything is to your satisfaction.

# Welcome to JavaScript community

[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### **Describe your change:**

* [ V] Add an algorithm?

### **Checklist:**
* [V] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
* [V] This pull request is all my own work -- I have not plagiarized.
* [V] I know that pull requests will not be merged if they fail the automated tests.
* [V] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [V] All new JavaScript files are placed inside an existing directory.
* [ V All filenames should use the UpperCamelCase (PascalCase) style.  There should be no spaces in filenames.
     **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not

### CODE REVIEW

**Review by raklaptudirm — CHANGES_REQUESTED** [review]
I would suggest moving the tests into a separate `jest` file. See the `test` subdirectory for more details.

**Comment by raklaptudirm** [comment]
Sorry, I mistakenly closed it.

File: Search/SlidingWindow.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+*
+* Very useful for keeping track of a subset of data in an
+* array/string etc.
+*
+* Time Complexity: Best - O(n);
+*
+* Examples:
+* maxSubarraySum([1,2,5,2,8,1,5],2) // returns 10
+* maxSubarraySum([1,2,5,2,8,1,5],15) // returns null
+* maxSubarraySum([5,2,6,9],3) // returns 17
```
Comment (Maker-Mark) [inline Search/SlidingWindow.js:17]: This example won't work. Do you mean slidingWindow?

Comment (morMBen) [reply Search/SlidingWindow.js:17]: Ohh... yes I did meant to write slidingWindow :)
Do you want me to rewrite it and send the pull request again?

Comment (raklaptudirm) [reply Search/SlidingWindow.js:17]: Just add more commits to this pr.

Comment (morMBen) [reply Search/SlidingWindow.js]: I did the commit, hope it's all right now.

**Comment by raklaptudirm** [comment]
@morMBen Please move the tests as i mentioned.

**Comment by morMBen** [comment]
As you requested I moved the tests.

**Comment by raklaptudirm** [comment]
You forgot to use `standard.js`.

**Comment by morMBen** [comment]
You're right, I'm just getting used to using it right now.
Done :)

**Comment by raklaptudirm** [comment]
Seems like you still have some problems.

**Comment by morMBen** [comment]
I did a new pull request that should work, hopefully, it'll work now.

**Pull request closed without merging**